File: pset1/mario_less.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;
    
    //checks if n is between 1 and 8 after get n
    do 
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
    
    //create n lines
    for (int i = 0; i < n; i++)
    {
        //create columns
        for (int j = n - 1; j >= i; j--)
        {
            //print white spaces
            if (j > i)
            {
                printf(" ");
            }
            
            //print # diagonal
            if (j == i)
            {
                printf("#");
                
                //print # in the remaining spaces
                for (int a = i; a > 0; a--)
                {
                    printf("#");
                }
            }
        }
    
        //jump one line
        printf("\n");
    }
}

File: pset6/mario_less.py
from cs50 import get_int

# Checks if n is between 1 and 8 and re-prompt if it's not
while True:
    n = get_int("Height: ")
    if n >= 1 and n <= 8:
        break
    
# Create n lines
for i in range(n):
    
    # Create columns
    for j in range(n - 1, i - 1, -1):
        # Print white spaces
        if j > i:
            print(" ", end="")
        
        if j == i:
            # Print # diagonal
            print("#", end="")
            # Print # in the remaining spaces and jump one line
            print("#" * i)